src/Compilers/Core/Portable/Optional.cs

```diff
@@ -56,5 +56,15 @@ public T Value
         {
             return new Optional<T>(value);
         }
+
+        /// <summary>
+        /// Returns a string representation of the optional value.
+        /// </summary>
+        public override string ToString()
+        {
+            return _hasValue
+                ? _value.ToString()
```

Comment:
I propose (and have pushed a revision with) the language "meaningful".  Essentially, all this type does is associate a flag with a value indicating whether that value is intended to be read (e.g. as in `TryParse` or `TryGetValue`).